service/worker/scheduler/workflow.go

```diff
@@ -1002,6 +1005,43 @@ func (s *scheduler) getListInfo(inWorkflowContext bool) *schedpb.ScheduleListInf
 	}
 }
 
+func (s *scheduler) updateCustomSearchAttributes(searchAttributes *commonpb.SearchAttributes) {
+	// We want to distinguish nil value from an empty object value.
+	// When it is nil, then it's a no-op. Otherwise, it will overwrite the search attributes.
+	// That is, if it is an empty map, it will unset all search attributes.
+	if searchAttributes == nil {
+		return
+	}
+
+	upsertMap := map[string]any{}
+	//nolint:staticcheck // SA1019 Use untyped version for backwards compatibility.
+	currentSearchAttributes := workflow.GetInfo(s.ctx).SearchAttributes
+	for key, currentValuePayload := range currentSearchAttributes.GetIndexedFields() {
+		if searchattribute.IsReserved(key) {
```

Comment:
possible determinism concern: if this returns false in one version, then we add a new attribute to the reserved set and this returns true, it may upsert something where it didn't before and break on replay. what do you think? I really don't want to use a sideeffect just for this... should we just call upsert with something we know is reserved and let it fail on the server?

Reply:
Actually, the server doesn't fail `UpsertSearchAttributes` for all reserved search attributes. It fails only for system ones. If the user tries to write to `BuildIds` for example, it would work...

Reply:
I think this is more of a general issue in the Server. Basically, if a custom search attribute name becomes a reserved search attribute (system, or predefined), the Server behavior is not well-defined, and likely to break the workflow if it needs to be replayed.

Reply:
Bleh :(  Ok, I guess we can't fix it all here. This should be rare enough (and we can write release notes) to say that users need to manually recreate schedules if they use a search attribute that becomes a reserved one in a new release.

Can you write a comment here saying that this is a deliberate determinism violation and explain when it may break and what we/users should do when it (inevitably) happens?